### PR TITLE
Pave the way for per-version transition guides

### PR DESCRIPTION
**Pull request by jougs**
This PR removes all transition information from anywhere but the transition guides. It also moves things around, so the transition guide can now be in a directory corresponding to the name of the Git tag under `release_notes`.

### CODE REVIEW

**Comment by jessica-mitchell** [comment]
thanks for this PR @jougs 
I was wondering whether or not the NEST 3.0 features should be made more visible - simply because these are substantial new additions to NEST; a user who has 3.1 might need to reference the node or connection handling guides. I assume that the reference guide comparing NEST 2.x to 3.0 is also valid for 3.x. I don't think 3.0 release notes would necessarily be the first place a user would look for such information.

What do you think of moving these to the guides/ directory, remove the 3.0 specification but also link to them from the release notes section for NEST 3.0?

**Comment by jougs** [comment]
I totally agree that the feature information from the NEST 3.0 transition guides needs to be integrated into the existing guides and probably new ones on node management and parametrization. Contrary to what I said in our video call earlier, that integration looks not so straightforward after all. I would thus postpone that to another PR and keep the scope of this one to what it currently is.

File: doc/userdoc/release_notes/v3.0/index.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+crypotgraphic generators. Most notably, you can now much more easily seed all generators and change the type of random
+number generator you are using.
+
+We have improved how :doc:`recordings from simulations <features/recording_simulations>` are handled, making the
+infrastructure more modular and extensible. In addition to the previously supported recording methods, a new backend for
+SIONlib is now available. The interface for :doc:`stimulation devices <features/stimulation_backends>` can now handle
+data from external sources, such as other simulators.
+
+The Topology Module is no longer a separate module; it is integrated within PyNest, and now referred to as support for
+:doc:`spatially-structured networks <../../guides/spatial/index>`.
```
Comment (jessica-mitchell) [inline doc/userdoc/release_notes/v3.0/index.rst:24]:
```suggestion
:doc:`spatially-structured networks <../../guides/spatially_structured_networks>`.
```

File: doc/userdoc/tutorials/pynest_tutorial/part_4_spatially_structured_networks.rst [hunk 2]
```diff
@@ -49,7 +49,7 @@ models that incorporate spatial location and spatial connectivity
 profiles.
 
 .. note::
-   For comprehensive documentation of spatial properties and connectivity, see the :doc:`../../guides/spatial/guide_spatially_structured_networks` [1]_.
+   For comprehensive documentation of spatial properties and connectivity, see the :doc:`../../guides/spatial/guide_spatially_structured_networks`.
```
Comment (jessica-mitchell) [inline doc/userdoc/tutorials/pynest_tutorial/part_4_spatially_structured_networks.rst:52]:
```suggestion
   For comprehensive documentation of spatial properties and connectivity, see the :doc:`../../guides/spatially_structured_networks`.
```

File: doc/userdoc/release_notes/v3.0/features/node_handles.rst [hunk 3]
```diff
@@ -185,7 +185,7 @@ Direct attributes
     >>> neurons.C_m
         (111.0, 111.0, 111.0, 111.0, 111.0, 111.0, 111.0, 111.0, 111.0, 111.0)
 
-    If your nodes are spatially distributed (see :ref:`spatially distributed nodes <topo_changes>`),
+    If your nodes are spatially distributed (see :doc:`../../../guides/spatial/index`),
```
Comment (jessica-mitchell) [inline doc/userdoc/release_notes/v3.0/features/node_handles.rst:188]:
```suggestion
    If your nodes are spatially distributed (see :doc:`../../../guides/spatially_structured_networks`),
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
@jougs if you want to keep this PR simple, I'm fine with that for now

I just noticed some broken links - can you check, I think there are other places in userdoc/ that the file name and location have not been updated for spatially_structured_networks

**Review by jessica-mitchell — APPROVED** [review]
thanks! lgtm

**Comment by jougs** [comment]
@terhorstd: I will merge without waiting for your review, as this blocks some other work of mine. If you have additional ideas, please add a comment here.

**Pull request merged**